app/main/lib/image_similarity.py

```diff
@@ -64,34 +65,48 @@ def add_image(save_params):
 def callback_add(task):
     return media_crud.add(task, ImageModel, ["pdq", "phash"])
 
-def blocking_search_image(task):
-    image, temporary, context, presto_result = media_crud.get_blocked_presto_response(task, ImageModel, "image")
-    threshold = task.get("threshold", 0.0)
-    limit = task.get("limit", 200)
-    model = app.config['IMAGE_MODEL']
+def search_image(image, model, limit, threshold, task, hash_value, context, temporary):
     if image:
         if model and model.lower() == "pdq":
             app.logger.info(f"Searching with PDQ.")
-            image.pdq = presto_result.get("body", {}).get("result", {}).get("hash_value")
-            result = search_by_pdq(image.pdq, threshold, context[0], limit)
+            image.pdq = hash_value
+            result = search_by_pdq(image.pdq, threshold, context, limit)
         else:
             app.logger.info(f"Searching with phash.")
-            image.phash = presto_result.get("body", {}).get("result", {}).get("hash_value")
-            result = search_by_phash(image.phash, threshold, context[0], limit)
+            image.phash = hash_value
+            result = search_by_phash(image.phash, threshold, context, limit)
         if temporary:
             media_crud.delete(task, ImageModel)
         else:
             media_crud.save(image, ImageModel, ["pdq", "phash"])
-        if task.get("limit"):
-            return {"result": result[:task.get("limit")]}
+        if limit:
+            return {"result": result[:limit]}
         else:
             return {"result": result}
     else:
         return {"error": "Image not found for provided task", "task": task}
 
+def blocking_search_image(task):
+    image, temporary, context, presto_result = media_crud.get_blocked_presto_response(task, ImageModel, "image")
+    threshold = task.get("threshold", 0.0)
```

Comment:
We've historically applied 0.0 throughout alegre - I think it's safe to continue applying that in this case but @computermacgyver should weigh in!

Reply:
`0` is all images in the context. In practice we're always including a threshold when Check API uses the code. I think `0` is a good default if a threshold is not given. My own concern would be whether we error out because the response is too large or something. I think, however, we're fine at present